### PR TITLE
ReaderLink: add location to stack on opening book

### PR DESCRIPTION
**Pull request by yparitcher**
The inital location is useful to go back to.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7685)

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]
I probably just overlooked that. :+1:

**Comment by poire-z** [comment]
Mhhh, I'm not sure, this wouldn't work for me :/
I open a book, read on - then a link, follow, possibly follow more links - then swipe to go back, swipe to go back again and expect the "Location history is empty" notification, which mean I'm back to my normal reading position.
With this, I'll be back to my opening position, with _no way_ to get forward to my reading position, that is lost at this point.

**Comment by yparitcher** [comment]
I switch books constantly, and then move around in the new one a little and then want to go back, I almost never use links, mostly chapter switching which will also trigger this

**Comment by yparitcher** [comment]
should i maybe hide it behind a pref?

**Comment by poire-z** [comment]
Dunno, it feels like an odd use case (and not much to deserve a menu item :/)
And the `Links>` menu is already full with 10 items.

I think it's really dangerous when one has "Swipe to go back" enabled. May be it hurts less when people don't, and would need to use the menu item? But they could also use another gesture for that.
May be if you can detect this ? (But I guess you have a gesture to go back :/)

Or an hidden setting ? (I added a few for my own strange needs...)

**Comment by poire-z** [comment]
Or may be in `Document>` (or `Navigation>`?): 
`Add opening page in location history` or something?

**Comment by NiLuJe** [comment]
+1 for Navigation, where there already a setting that controls something that interacts with the location stack (related to the Back button/key), IIRC?

File: frontend/ui/elements/common_settings_menu_table.lua [hunk 1]
```diff
@@ -436,6 +436,15 @@ if Device:hasKeys() then
         end,
     }
 end
+common_settings.opening_page_location_stack = {
+        text = _("Add opening page in location history"),
```
Comment (poire-z) [inline frontend/ui/elements/common_settings_menu_table.lua:440]: `Add ...*to* location history` ?

**Review by poire-z — APPROVED** [review]

**Pull request merged**